### ISSUE TITLE
Empty cooperation-conditions element

### ISSUE THREAD

**Issue by jiripetrzelka** [erasmus-without-paper/ewp-specs-api-iias#146]
What is the expected behaviour of clients in case they receive this in an IIA?

`<cooperation-conditions/>`

As I understand it, this is not against the specification because minOccurs="1" for this element is satisfied and minOccurs="0" for all possible sub-elements too.

What are clients expected to do then?
- Report it as an error to the Stats portal and not show it to their end users?
- Not report it to the Stats portal and show it to the end users as a let them proceed with an approval?
- Not report it to the Stats portal and show it to the end users as a draft and not allow them to approve the IIA (despite the conditions-hash being present)?

My opinion is that such IIAs should be forbidden by the specification to appear in the network and that the first approach described above should be taken.

Second issue related to this matter: If the IIA has been terminated as a whole, are server implementers:
- Required to keep the latest cooperation-conditions mutually approved and add the terminated-as-a-whole attribute?
- Required to remove all sub-elements of the cooperation-conditions element, i.e. publish `<cooperation-conditions terminated-as-a-whole="true"/>`?
- Allowed to do both and expect clients to interpret both approaches as equal?

**Comment by demilatof** [comment]
> Second issue related to this matter: If the IIA has been terminated as a whole, are server implementers:
> 
>- Required to keep the latest cooperation-conditions mutually approved and add the terminated-as-a-whole attribute?
> - Required to remove all sub-elements of the cooperation-conditions element, i.e. publish `<cooperation-conditions terminated-as-a-whole="true"/>`?
>- Allowed to do both and expect clients to interpret both approaches as equal?

Personally, I'd like better the first option to have both of the information in the last snapshot (the one after we approve terminated-as-a-whole):

1. The cooperation conditions initially approved
2. The fact that the IIA has been terminated as a whole

Your second option could become impossible if they state that it's forbidden to have empty CCs.
Anyway, both of the partners should do the same thing.
Let's wait what they will answer you.

**Comment by janinamincer-daszkiewicz** [comment]
> My opinion is that such IIAs should be forbidden by the specification to appear in the network and that the first approach described above should be taken.

I agree. Only those IIAs should appear in the network which are ready to be approved. IIA with empty cooperation conditions does not belong to this category.

**Comment by janinamincer-daszkiewicz** [comment]
> Personally, I'd like better the first option to have both of the information in the last snapshot (the one after we approve terminated-as-a-whole):

Me too. We decided to add the extra element to tag IIA as terminated as a whole in order to be able to keep the rest of IIA intact.

**Comment by demilatof** [comment]
> > Personally, I'd like better the first option to have both of the information in the last snapshot (the one after we approve terminated-as-a-whole):
> 
> Me too. We decided to add the extra element to tag IIA as terminated as a whole in order to be able to keep the rest of IIA intact.

When you say "the rest of IIA intact" what do you mean with "the rest"?
Since it is terminated as a whole should not be the whole agreement, not only a part?
If it is a part, what part of the agreement?

**Comment by janinamincer-daszkiewicz** [comment]
I mean cooperation conditions section.

**Comment by demilatof** [comment]
> I mean cooperation conditions section.

Thanks, therefore all the mobilities inside the cooperation conditions section will remain the same before and after the "terminate-as-a-whole"

**Comment by jiripetrzelka** [comment]
@janinamincer-daszkiewicz So do you plan to add for example this sentence into the specification?
"The cooperation-conditions element must contain at least one sub-element."

**Comment by janinamincer-daszkiewicz** [comment]
There will be no more changes in the specification. Version 7 is frozen.
